protocol/test/forge/v2/safeGuards/ThresholdSafeGuard.t.sol

```diff
@@ -277,165 +546,207 @@ contract ThresholdSafeGuardTest is ThresholdSafeGuardTestBase {
-        bytes memory signature = bytes.concat(
-            signexecutor(safeOwners[0]), 
-            signEOA(dataHash, pk2), 
-            signEOA(dataHash, pk3)
-        );
 
         // Check with a different executor to what we signed with
-        vm.expectRevert("GS025");
-        doCheck("", 0, safeOwners[1], signature);
+        checkTransaction(
+            ExpectSuccessOrFailure.FAILURE_INCORRECT_EXECUTOR,
+            safeOwners[1], // Not the same as the AS_EXECUTOR below
+            makeTransactionParams(
+                0, 
+                "", 
+                makeSignSpecs(
+                    SignSpec(SignType.AS_EXECUTOR, safeOwners[0]),
+                    SignSpec(SignType.AS_EOA_TX, safeOwners[1]),
+                    SignSpec(SignType.AS_EOA_TX, safeOwners[2])
+                )
+            )
+        );
     }
 
     function test_checkTransaction_withOverrideTresholdSet_zeroSelector() public {
         vm.startPrank(executor);
+        // Magically this function is hashed to `0x00000000`
         bytes4 fnSelector = bytes4(keccak256("wycpnbqcyf()"));
         assertEq(fnSelector, bytes4(0));
 
         guard.setFunctionThreshold(address(mock), fnSelector, 3);
 
-        // Add a couple more executors to the safe which we can sign on behalf of.
-        (uint256 pk2, uint256 pk3) = addSafeOwners();
+        // Fails with 2 signers
+        checkTransaction(
+            ExpectSuccessOrFailure.FAILURE_THRESHOLD,
+            safeOwners[0], 
+            makeTransactionParams(
+                100e18, 
+                abi.encodeWithSelector(fnSelector), 
+                makeSignSpecs(
+                    SignSpec(SignType.AS_EXECUTOR, safeOwners[0]),
+                    SignSpec(SignType.AS_EOA_TX, safeOwners[1])
+                )
+            )
+        );
 
-        // An zero selector
-        bytes memory fnCall = abi.encodeWithSelector(fnSelector);
-        bytes32 dataHash = getTxHash(fnCall, 100e18);
-        bytes memory signature = bytes.concat(
-            signexecutor(safeOwners[0]),
-            signEOA(dataHash, pk2)
+        // OK with 3 signers
+        checkTransaction(
+            ExpectSuccessOrFailure.SUCCESS,
+            safeOwners[0], 
+            makeTransactionParams(
+                100e18, 
+                abi.encodeWithSelector(fnSelector), 
+                makeSignSpecs(
+                    SignSpec(SignType.AS_EXECUTOR, safeOwners[0]),
+                    SignSpec(SignType.AS_EOA_TX, safeOwners[1]),
+                    SignSpec(SignType.AS_EOA_TX, safeOwners[2])
+                )
+            )
         );
+    }
 
-        // Fails with 2 signers
-        vm.expectRevert("!Dynamic Signature Threshold");
-        doCheck(fnCall, 100e18, safeOwners[0], signature);
+    function test_checkTransaction_withOverrideThresholdSet() public {
+        vm.startPrank(executor);
+        bytes4 fnSelector = bytes4(keccak256("doThing(string,uint256)"));
```

Comment:
nit: could make these variables global

Reply:
Done in https://github.com/TempleDAO/temple/pull/842/commits/c15b3684c8653f0ef6804836d08de11920e85f81